### PR TITLE
Publish all rocks job

### PR DESCRIPTION
**Pull request by gboutry**
Create the workflow `publish_branch`, allowing any maintainer to trigger the build of all the ROCKs of a selected branch. Limited to branches: main, stable/2023.1

### CODE REVIEW

File: .github/workflows/publish_branch.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - uses: canonical/craft-actions/rockcraft-pack@main
+        id: rockcraft
+        with:
+          path: rocks/${{ matrix.rock }}
+      - uses: actions/upload-artifact@v3
```
Comment (hemanthnakkina) [inline .github/workflows/publish_branch.yaml:60]: I am just thinking whether we can avoid upload here and download again in publish step.

Comment (gboutry) [reply .github/workflows/publish_branch.yaml:60]: Yeah, I'll make publish a step, and comply to the matrix

File: .github/workflows/publish_branch.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        id: rockcraft
+        with:
+          path: rocks/${{ matrix.rock }}
+      - name: Install skopeo
+        run: |
+          sudo snap install --devmode --channel edge skopeo
+      - name: Install yq
+        run: |
+          sudo snap install yq
+      - uses: actions/download-artifact@v3
```
Comment (hemanthnakkina) [inline .github/workflows/publish_branch.yaml:78]: this step is not required

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]

File: .github/workflows/publish_branch.yaml (hunk 2 above)
Comment (gboutry) [reply .github/workflows/publish_branch.yaml:78]: Is YQ part of the runner ?

Comment (hemanthnakkina) [reply .github/workflows/publish_branch.yaml:78]: YQ is not part of runner, the question is about download-artifact

**Review by hemanthnakkina — APPROVED** [review]

**Pull request merged**